pyecm.py

```diff
@@ -1,1483 +1,1569 @@
-   def mpz(n):
-      '''A dummy function to ensure compatibility with those that do not have gmpy.
+        while 1:
+            for prime in PRIMES:
+                if not n % prime:
+                    should_break = True
+                    break
 
-Returns n.'''
-      return n
+            if should_break:
+                should_break = False
+                n += inc
+                inc = 6 - inc
+                continue
 
-   def root(n, k):
-      '''Finds the floor of the kth root of n. This is a duplicate of gmpy's root function.
+            p = 1
+            for i in range(int(math.log(n) / LOG_2), 0, -1):
+                p <<= (n >> i) & 1
+                p = (p * p) % n
 
-Returns a tuple. The first item is the floor of the kth root of n. The second is 1 if the root is exact (as in, sqrt(16)) and 0 if it is not.'''
-      low = 0
-      high = n + 1
-      while high > low + 1:
-         mid = (low + high) >> 1
-         mr = mid**k
-         if mr == n:
-            return (mid, 1)
-         if mr < n:
-            low = mid
-         if mr > n:
-            high = mid
-      return (low, 0)
+            if p == 1:
+                return n
 
-   def sqrt(n):
-      return root(n, 2)[0]
+            n += inc
+            inc = 6 - inc
+
+    def mpz(number):
+        """
+        A dummy function to ensure compatibility with those that do not have gmpy.
+        Returns n.
+        """
+        return number
+
+    def root(n, k):
+        """
+        Finds the floor of the kth root of n. This is a duplicate of gmpy's root function.
+        Returns a tuple. The first item is the floor of the kth root of n.
+        The second is 1 if the root is exact (as in, sqrt(16)) and 0 if it is not.
+        """
+        low = 0
+        high = n + 1
+        while high > low + 1:
+            mid = (low + high) >> 1
+            mr = mid ** k
+            if mr == n:
+                return mid, 1
+            if mr < n:
+                low = mid
+            if mr > n:
+                high = mid
+        return low, 0
+
+    def sqrt(n):
+        return root(n, 2)[0]
 
 # We're done importing. Now for some constants.
 if GMPY_EXISTS:
-   INV_C = 1.4
+    INV_C = 1.4
 else:
-   if PSYCO_EXISTS:
-      INV_C = 7.3
-   else:
-      INV_C = 13.0
+    INV_C = 13.0
 LOG_2 = math.log(2)
 LOG_4 = math.log(4)
 LOG_3_MINUS_LOG_LOG_2 = math.log(3) - math.log(LOG_2)
 LOG_4_OVER_9 = LOG_4 / 9
 _3_OVER_LOG_2 = 3 / LOG_2
 _5_LOG_10 = 5 * math.log(10)
 _7_OVER_LOG_2 = 7 / LOG_2
-BIG = 2.0**512
-BILLION = 10**9 # Something big that fits into an int.
+BIG = 2.0 ** 512
+BILLION = 10 ** 9  # Something big that fits into an int.
 MULT = math.log(3) / LOG_2
 ONE = mpz(1)
-SMALL = 2.0**(-30)
+SMALL = 2.0 ** (-30)
 SMALLEST_COUNTEREXAMPLE_FASTPRIME = 2047
 T = (type(mpz(1)), type(1), type(1))
-DUMMY = 'dummy' # Dummy value throughout the program
-VERSION = '2.0.5 (Python 3)'
+DUMMY = 'dummy'  # Dummy value throughout the program
+VERSION = '3.0.0'
```

Comment:
Please no version changes here. Version changes occur when I choose to do a release, not on every commit. Plus, there's no reason to bump the major version here.